lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/SecurityTokenHelper.java

```diff
@@ -0,0 +1,122 @@
+ */
+package org.apache.reef.bridge.client;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.Token;
+import javax.xml.bind.DatatypeConverter;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Helper class to provide security token methods.
+ */
+final class SecurityTokenHelper {
+
+  private static final Logger LOG = Logger.getLogger(YarnJobSubmissionClient.class.getName());
+
+  private SecurityTokenHelper() {
+  }
+
+  /**
+   * Get list of TokenInfo.
+   *
+   * @param securityTokenIdentifierFile
+   * @param securityTokenPasswordFile
+   * @param tokenKindString
+   * @return
+   * @throws IOException
+   */
+  static List<TokenInfo> getTokenInfo(final String securityTokenIdentifierFile,
+                                              final String securityTokenPasswordFile,
+                                              final String[] tokenKindString) throws IOException {
+    final List<TokenInfo> tokens = new ArrayList<>();
+    for (final String tokenStr : tokenKindString) {
+      LOG.log(Level.INFO, "tokenKind for each token: " + tokenStr);
+      final String[] tokenParts = tokenStr.split(",");
+      if (tokenParts.length != 2) {
+        LOG.log(Level.SEVERE, "Expected token parts length is 2 with format [tokenKind, tokenLen].");
+        throw new IllegalArgumentException("Expected token parts length is 2");
+      }
+
+      tokens.add(new TokenInfo(tokenParts[0], tokenParts[0], Integer.parseInt(tokenParts[1])));
+    }
+
+    FileInputStream fStream = new FileInputStream(securityTokenPasswordFile);
+    BufferedReader reader = new BufferedReader(new InputStreamReader(fStream, "UTF-8"));
+    for (final TokenInfo info : tokens) {
+      info.setPassword(reader.readLine());
+    }
+    reader.close();
+
+    FileInputStream in = new FileInputStream(securityTokenIdentifierFile);
+    DataInputStream din = new DataInputStream(in);
+    for (final TokenInfo info : tokens) {
+      final int len = info.getLength();
+      final byte[] b = new byte[len];
+      final int lenRead = din.read(b, 0, len);
+      LOG.log(Level.INFO, "$$$$$$$ token length read: " + lenRead);
+      info.setKey(b);
+    }
+    in.close();
+    din.close();
+    return tokens;
+  }
+
+  /**
+   * Add a token to the currentUser.
+   * Log info for debug purpose for now until it is stable then we can change the log level to FINE.
+   *
+   * @param user
+   * @param identifier
+   * @param password
+   * @param tokenKind
+   * @param tokenSerive
+   */
+  static void addToken(final UserGroupInformation user,
+                               final byte[] identifier,
+                               final byte[] password,
+                               final Text tokenKind,
+                               final Text tokenSerive) {
+    LOG.log(Level.INFO, "addToken for " + tokenKind);
```

Comment:
here and everywhere in logging: use string interpolation, think of different logging levels